lms/djangoapps/courseware/courses.py

```diff
@@ -415,3 +415,29 @@ def get_studio_url(course, page):
     if is_studio_course and is_mongo_course:
         studio_link = get_cms_course_link(course, page)
     return studio_link
+
+
+def get_problems_in_section(section):
+    """
+    This returns a dict having problems in a section.
+    Returning dict has problem location as keys and problem
+    descriptor as values.
+    """
+
+    problem_descriptors = defaultdict()
+    if not isinstance(section, UsageKey):
+        section_key = UsageKey.from_string(section)
+    else:
+        section_key = section
+    # it will be a Mongo performance boost, if you pass in a depth=3 argument here
+    # as it will optimize round trips to the database to fetch all children for the current node
+    section_descriptor = modulestore().get_item(section_key, depth=3)
+
+    # iterate over section, sub-section, vertical
+    for subsection in section_descriptor.get_children():
```

Comment:
`modulestore.get_items()` fetch items under a course whereas we are looking for items under a chapter or section so it would not serve the purpose here.